testing/logging/test_formatter.py

```diff
@@ -34,3 +36,31 @@ class option(object):
     formatter = ColoredLevelFormatter(tw, logfmt)
     output = formatter.format(record)
     assert output == ("dummypath                   10 INFO     Test Message")
+
+
+@pytest.mark.skipif(
+    six.PY2, reason="Formatter classes don't support format styles in PY2"
+)
+def test_multiline_message():
+    from _pytest.logging import PercentStyleMultiline
+
+    logfmt = "%(filename)-25s %(lineno)4d %(levelname)-8s %(message)s"
+
+    record = logging.LogRecord(
+        name="dummy",
+        level=logging.INFO,
+        pathname="dummypath",
+        lineno=10,
+        msg="Test Message line1\nline2",
+        args=(),
+        exc_info=False,
+    )
+    # this is called by logging.Formatter.format
+    record.message = record.getMessage()
+
+    style = PercentStyleMultiline(logfmt)
+    output = style.format(record)
+    assert output == (
+        "dummypath                   10 INFO     Test Message line1\n"
+        "dummypath                   10 INFO     line2"
```

Comment:
Hmm, I think it would be better for this to just be indented, but not have the prefix for the second line - it is rather confusing, since it looks like two separate log messages (although there is an indicator (`\n`)).

Reply:
I also think it would be better to just indent this line. The problem is that it is not so easy to determine the indentation in a generic way.  A year ago I wrote [this](https://github.com/thisch/pytest/commit/4c846c8559bae7e9322d670928e5e90acbc02cfe), which  I don't really like.

Reply:
Here is a simplified approach of my commit from last year:

```diff
diff --git a/src/_pytest/logging.py b/src/_pytest/logging.py
index 1f2d5e1d4..70ce15ad9 100644
--- a/src/_pytest/logging.py
+++ b/src/_pytest/logging.py
@@ -17,6 +17,7 @@ from _pytest.pathlib import Path
 
 DEFAULT_LOG_FORMAT = "%(levelname)-8s %(name)s:%(filename)s:%(lineno)d %(message)s"
 DEFAULT_LOG_DATE_FORMAT = "%H:%M:%S"
+_ESCAPE_SEQ = re.compile(r'\x1b\[[\d;]+m')
 
 
 class ColoredLevelFormatter(logging.Formatter):
@@ -88,15 +89,13 @@ if not six.PY2:
 
         def format(self, record):
             if "\n" in record.message:
-                fmt = self._fmt
-                dct = record.__dict__
-
-                data_seq = [
-                    # fmt % {**dct, "message": line}
-                    fmt % self._update_message(dct, line)
-                    for line in record.message.splitlines()
-                ]
-                return "\n".join(data_seq)
+                lines = record.message.splitlines()
+                formatted = self._fmt % self._update_message(
+                    record.__dict__, lines[0])
+                indentation = (_ESCAPE_SEQ.sub('', formatted)
+                               .find(lines[0]))
+                lines[0] = formatted
+                return ("\n" + ' '*indentation).join(lines)
             else:
                 return self._fmt % record.__dict__
 
diff --git a/testing/logging/test_formatter.py b/testing/logging/test_formatter.py
index 2fa4b3157..0879875df 100644
--- a/testing/logging/test_formatter.py
+++ b/testing/logging/test_formatter.py
@@ -62,5 +62,5 @@ def test_multiline_message():
     output = style.format(record)
     assert output == (
         "dummypath                   10 INFO     Test Message line1\n"
-        "dummypath                   10 INFO     line2"
+        "                                        line2"
     )
```

Reply:
Doesn't look too bad to me..!

FWIW: I'm using loguru for my current project, where log messages end up on stdout/stderr instead - should check how it behaves with regard to multiple lines there.